tests/Http/Three/PostTest.php

```diff
@@ -553,6 +584,24 @@ public function testPostsIndexAsAdmin()
         ]);
     }
 
+    /**
+     * Test for retrieving all posts without the required scope.
+     *
+     * GET /api/v3/posts
+     * @return void
+     */
+    public function testPostsIndexWithoutRequiredScope()
+    {
+        // Admins should see all posts.
+        factory(Post::class, 'accepted', 10)->create();
+        factory(Post::class, 'rejected', 5)->create();
+
+        $response = $this->getJson('api/v3/posts');
```

Comment:
Oh hmm, I think there's a catch here! 🎣 We want to make sure this route is publicly accessible (for example, so that anonymous users visiting a campaign page can see a gallery before signing up).

I'm thinking that we only want Gateway's `scope` middleware to check scopes at all if there isn't a token... it feels like they're irrelevant in cases where we allow anonymous traffic. What do you think about adding a conditional like this to the [`scope` middleware](https://github.com/DoSomething/gateway/blob/00e3e567f0ab8fdbf85c75c5d4a1b894427120ab/src/Server/Middleware/RequireScope.php#L38)?

```php
    public function handle($request, Closure $next, ...$requestedScopes)
    {
        if (! $this->token->exists()) {
            return $next($request);
        }

        $providedScopes = $this->token->scopes;
        // ...
```

Reply:
@DFurnes that makes sense to me! And then edit all the tests so that anonymous users _can_ see all the `GET` v3 endpoints?

Reply:
@DFurnes just bumping this in case you missed it! I _think_ that all `GET` v3 endpoints should be ok for anonymous users to see but wanted to double check!

Reply:
Yeah, I think that sounds good! 👍